sycl/doc/SYCLCompilerAndRuntimeDesign.md

```diff
@@ -102,16 +102,18 @@ pointers to the device memory. As there is no way in OpenCL to pass structures
 with pointers inside as kernel arguments all memory objects shared between host
 and device must be passed to the kernel as raw pointers.
 SYCL also has a special mechanism for passing kernel arguments from host to
-the device. In OpenCL you need to call `clSetKernelArg`, in SYCL all the
-kernel arguments are captures/fields of lambda/functor SYCL functions for
-invoking kernels (such as `parallel_for`). For example, in the previous code
-snippet above `accessor` `A` is one such captured kernel argument.
-
-To facilitate the mapping of the captures/fields of lambdas/functors to OpenCL
-kernel and overcome OpenCL limitations we added the generation of an OpenCL
+the device. In OpenCL kernel arguments are set by calling `clSetKernelArg` function
+for each kernel argument, meanwhile in SYCL all the kernel arguments are fields of
+"SYCL kernel function" which can be defined as a lambda function or a named function
+object and passed as an argument to SYCL function for invoking kernels (such as
+`parallel_for` or `single_task`). For example, in the previous code snippet above
+`accessor` `A` is one such captured kernel argument.
+
+To facilitate the mapping of the fields of SYCL kernel functions to OpenCL
```

Comment:
```suggestion
To facilitate the mapping of SYCL kernel data members to OpenCL
```
"fields of .. functions" doesn't make sense. SYCL kernel is not a function it's a function object.

Reply:
It was motivated by the spec. See Glossary section for example.

> SYCL kernel function A type which is callable with ‘operator()‘ that takes a id, item, nd-item or group which can be passed to kernel enqueue member functions of the command group handler. A SYCL kernel function defines an entry point to a kernel. The function object can be a named standard layout type or lambda function.

But you are right, It sounds strange for me too...